examples/ecs/observers.rs

```diff
@@ -0,0 +1,199 @@
+//! This examples illustrates several ways you can employ observers
+
+use bevy::{
+    prelude::*,
+    utils::{HashMap, HashSet},
+};
+
+fn main() {
+    App::new()
+        .add_plugins(DefaultPlugins)
+        .add_systems(Startup, setup)
+        .add_systems(Update, (draw_shapes, handle_click))
+        .run();
+}
+
+#[derive(Component)]
+struct Mine {
+    pos: Vec2,
+    size: f32,
+}
+
+#[derive(Component)]
+struct TriggerMines {
+    pos: Vec2,
+    radius: f32,
+}
+
+#[derive(Component)]
+struct Explode;
+
+fn setup(world: &mut World) {
+    world.spawn(Camera2dBundle::default());
+
+    let font = world
+        .resource::<AssetServer>()
+        .load("fonts/FiraMono-Medium.ttf");
+    world.spawn(TextBundle::from_section(
+        "Click on a \"Mine\" to trigger it.\n\
+            When it explodes it will trigger all overlapping mines.",
+        TextStyle {
+            font,
+            font_size: 24.,
+            color: Color::WHITE,
+        },
+    ));
+
+    // Pre-register all our components, resource and event types.
```

Comment:
The important part here is that the ids assigned to the "primitive" events are consistent so we don't have to look them up via type id in the hot path.

This will continue to be the case for things like wildcard components, cleanup policy components etc. in a world with relations.

Where this code is in the codebase I'm totally open to.

Reply:
Yup this was my reaction as well. This is a major UX hiccup that we should fill in.

Reply:
This is fixed now.